includes/sanitizers/class-amp-core-theme-sanitizer.php

```diff
@@ -1522,4 +1569,446 @@ public function add_twentyfourteen_search() {
+
+			// If the toggle target is 'next' ir a sub-menu, only give the clicked toggle the active class.
+			if ( 'next' === $toggle_target || $this->has_class( $target_node, 'sub-menu' ) ) {
+				$this->add_amp_action( $toggle, 'tap', "{$toggle_id}.toggleClass(class='active')" );
+			} else {
+				// If not, toggle all toggles with this toggle target.
+				$target_toggles = $this->xpath->query( "//*[ @data-toggle-target = '{$toggle_target}' ]" );
+				foreach ( $target_toggles as $target_toggle ) {
+					if ( $this->has_class( $target_toggle, 'close-nav-toggle' ) ) {
+						// Skip adding the 'active' class on the "Close" button in the primary nav menu.
+						continue;
+					}
+					$target_toggle_id = $this->get_element_id( $target_toggle );
+					$this->add_amp_action( $toggle, 'tap', "{$target_toggle_id}.toggleClass(class='active')" );
+				}
+			}
+
+			// Toggle body class.
+			if ( $toggle->hasAttribute( 'data-toggle-body-class' ) ) {
+				$body_class = $toggle->getAttribute( 'data-toggle-body-class' );
+				$this->add_amp_action( $toggle, 'tap', "{$body_id}.toggleClass(class='{$body_class}')" );
+			}
+
+			// Leaving these for now as the theme looks broken in non-AMP already.
+			// @todo Toggle aria-expanded on the toggle.
+
+			if ( $toggle->hasAttribute( 'data-set-focus' ) ) {
+				$focus_selector = $toggle->getAttribute( 'data-set-focus' );
+
+				if ( ! empty( $focus_selector ) ) {
+					$focus_xpath   = $this->xpath_from_css_selector( $focus_selector );
+					$focus_element = $this->xpath->query( $focus_xpath )->item( 0 );
+
+					// Instead of manually setting or unsetting the focus here, we're going
+					// with the autofocus attribute instead (AMP does not have a "blur" action).
+					// According to the HTML 5 spec, only one element should have "autofocus",
+					// however it seems like AMP has turned this into a common mechanism for
+					// focusing on inputs after a lightbox (i.e. modal) opens.
+					if ( $focus_element instanceof DOMElement && in_array( $focus_element->tagName, [ 'input', 'select', 'textarea' ], true ) ) {
+						$focus_element->setAttribute( 'autofocus', true );
+					}
+				}
+			}
+		}
+	}
+
+	/**
+	 * Get the ID for an element.
+	 *
+	 * If the element does not have an ID, create one first.
+	 *
+	 * @param DOMElement $element Element to get the ID for.
+	 * @param string     $prefix  Optional. Defaults to '_amp_wp_id_'.
+	 * @return string ID to use.
+	 */
+	protected function get_element_id( $element, $prefix = 'amp-wp-id' ) {
+		static $index_counter = [];
+
+		if ( $element->hasAttribute( 'id' ) ) {
+			return $element->getAttribute( 'id' );
+		}
+
+		if ( ! array_key_exists( $prefix, $index_counter ) ) {
+			$index_counter[ $prefix ] = 2;
+			$element->setAttribute( 'id', $prefix );
+
+			return $prefix;
+		}
+
+		$id = "{$prefix}-{$index_counter[ $prefix ]}";
+		$index_counter[ $prefix ] ++;
+
+		$element->setAttribute( 'id', $id );
+
+		return $id;
+	}
```

Comment:
Why not use/polyfill [`wp_unique_id()`](https://developer.wordpress.org/reference/functions/wp_unique_id/)?

Reply:
Because I didn't know about it... :)

Also, it looks broken to me. We'll end up with `wp-amp-toggle-357` and `wp-amp-toggle-523`... It should keep count of the index per prefix to build clean IDs.